rotkehlchen/exchanges/binance.py

```diff
@@ -1205,15 +1365,34 @@ def _api_query_list_within_time_delta(
             else end_ts  # Case request without offset (1 request)
         )
         while True:
+            current_page = 1
             options = {
                 'timestamp': ts_now_in_ms(),
                 'startTime': from_ts,
                 'endTime': to_ts,
             }
             if additional_options:
                 options.update(additional_options)
-            result = self.api_query_list(api_type, method, options=options)
-            results.extend(result)
+                if method == 'lending/union/interestHistory':
+                    options['current'] = current_page
+
+            if method == 'lending/union/interestHistory':
+                last_queried_entry: tuple[str, int] = ('', 0)
```

Comment:
add a comment here that this check is to just protect against misbehaving binance giving us an infinite loop